notebooks/predicting_queries/simple_intent_suggestion.ipynb

```diff
@@ -0,0 +1,570 @@
+    "\n",
+    "query_taxons = queries_df.merge(taxons_df,on='pagePath')\n",
+    "# Split queries up (they're stupidily aggregated in the sql)\n",
+    "query_taxons['query']=query_taxons['search_terms'].map(lambda x: x.split(','))\n",
+    "query_taxons = query_taxons.explode('query')\n",
+    "# Average number of taxons visited per query (around 8)\n",
+    "np.mean(query_taxons.groupby('session_id').size())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Text preprocessing\n",
+    "translator = str.maketrans('', '', string.punctuation)\n",
+    "# Some multiword tokens\n",
+    "tokenizer = MWETokenizer([('log', 'in'), ('sign', 'in'), ('sign', 'up')])\n",
+    "# Tokenise query, explode, and regroup by taxon \n",
+    "# End up with a list of tokens per taxon\n",
+    "query_taxons['tokens'] = query_taxons['query'].map(lambda x: tokenizer.tokenize(x.translate(translator).split()))\n",
+    "query_taxons = query_taxons.explode('tokens')\n",
+    "query_taxons = query_taxons[['taxon','tokens']]\n",
+    "query_taxons = query_taxons.dropna()\n",
+    "# Group query tokens by taxon so we have a list of tokens per taxon\n",
+    "query_taxons = query_taxons.groupby('taxon').aggregate(lambda x: list(x)).reset_index()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# All the maintstream content text,titles from gov uk (assumed targets for users)\n",
+    "#title_df =graph.run(\"match (m:Mainstream) return m.name as name, m.title as title, m.text as text\").to_data_frame()\n",
+    "#title_df.to_csv(PROCESSED_DATA+'/mainstreamcontent.csv')\n",
+    "title_df = pd.read_csv(PROJECT_DIR+'/data/processed/mainstreamcontent.csv')\n",
+    "title_df = title_df.dropna()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#We want to weight titles a bit highly, so err just multiply by 3\n",
+    "title_df['body_tokens'] = title_df['text'] + ' ' + (title_df['title'] + ' ' * 3)\n",
+    "title_df['body_tokens'] = title_df['body_tokens'].map(lambda x: tokenizer.tokenize(x.translate(translator).split()))\n",
+    "title_df['body_tokens'] = title_df['body_tokens'].map(lambda x: [token.lower() for token in x])\n",
+    "title_df = title_df.dropna()\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "title_df.head(1)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Intersection of words in queries + content\n",
+    "service_terms= set(title_df['body_tokens'].explode('body_tokens'))\n",
+    "# Lose some stop words\n",
+    "intersection = service_terms.difference(set(['i','of','the','a','and','to','you','if','that','then','in','on']))"
```

Comment:
I guess you've considered more comprehensive ways to remove stopwords? Here's an example:
https://github.com/alphagov/govuk-entity-personalisation/blob/main/src/utils/preprocess.py#L43

Reply:
yeh wanted to think about what i was binning off, but doesnt make much difference really